### PR TITLE
Use multiple restoreSnapshotURLs on restore and general enhancements

### PR DESCRIPTION
**Pull request by philipwu08**

### CODE REVIEW

File: dr-script/migrate-hcp.sh [hunk 1]
```diff
@@ -185,7 +194,7 @@ EOF
 
     for POD in ${ETCD_PODS}; do
       # Create a pre-signed URL for the etcd snapshot
-      ETCD_SNAPSHOT="s3://${BUCKET_NAME}/${HC_CLUSTER_NAME}-${POD}-snapshot.db"
+      ETCD_SNAPSHOT="s3://${BUCKET_NAME}/${HC_CLUSTER_NAME}-etcd-0-snapshot.db"
```
[the remark this answers is not in the document]
Comment (philipwu08) [reply dr-script/migrate-hcp.sh:197]: Hey Mike, this is a temp workaround for our restore etcd problem.

On backup, we only backup etcd-0. On restore (old CRD), we're required to pass in the number of URLs that matches the number of pods, so we pass in etcd-0 for all of them.

When we move to the new CRD and new CPO controller, we will just pass in one URL, which is the etcd-0 image

File: dr-script/migrate-hcp.sh [hunk 2]
```diff
@@ -21,6 +21,12 @@ function get_hc_kubeconfig() {
     set -e
 }
 
+function get_hc_kubeadmin_pass() {
+  export KUBECONFIG="${MGMT_KUBECONFIG}"
+
+  HC_PASS=$(oc get secret -n  ${HC_CLUSTER_NS}-${HC_CLUSTER_NAME} kubeadmin-password -ojsonpath='{.data.password}' | base64 -d)
```
[the remark this answers is not in the document]
Comment (philipwu08) [reply dr-script/migrate-hcp.sh:27]: Right, so we have the option to pass in the HC_PASS. Only in the case where it's not pass, then we try to get it. It's really more of a convenience for testing. If the HC_PASS is incorrect, it'll fail early on in the backup when we create the configmap.

File: dr-script/migrate-hcp.sh [hunk 3]
```diff
@@ -517,43 +538,57 @@ function teardown_old_svc() {
...
 }
 
 REPODIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/.."
 
 source $REPODIR/dr-script/common.sh
 
-while getopts "i:n:s:p:" opt
+if [ "$#" -ne 2 ]; then
+  helpFunc
+  exit 1
```
Comment (mikeshng) [inline dr-script/migrate-hcp.sh:551]: @philipwu08 I am guessing this means those params are now mandatory.
I think we can update the common.sh and the example and the README to reflect this change if I read the code correctly.

Comment (philipwu08) [reply dr-script/migrate-hcp.sh:551]: Right Mike, run it like this now (no point to provide short arg if these two are min-required args):
`
./migrate-hcp.sh 27c308jdqnoud619dulgicckk8qjeu7d   phwu-stage-20
`

**Review by mikeshng — APPROVED** [review]
/approve

/lgtm

**Pull request merged**